src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -52,6 +59,15 @@ public StorageOperationException(String message) {
         }
     }
 
+    /**
+     * Signals that some error has occurred while trying to write data to the storage file.
+     */
+    public static class StorageOperationReadOnlyException extends StorageOperationException {
```

Comment:
@ngshikang 
This implementation of exception handling will still result in the program crashing. How would you catch the exception to prevent it from crashing? (Remember the cases we discussed in the tutorial)
Also, as discussed in the tutorial, the exception needs to be caught as near to its occurrence as possible.